sympy/solvers/solvers.py

```diff
@@ -1655,7 +1711,8 @@ def _expand(p):
                             # whose roots can be shown to be real with the
                             # unsimplified form of the solution whereas only one of
                             # the simplified forms appears to be real.
-                            flags['simplify'] = flags.get('simplify', False)
+                            if flags['simplify'] is None:
+                                flags['simplify'] = False
```

Comment:
This is another place that distinguished between `simplify` being `None` or `True`.